### PR TITLE
feat(server): verify domain record

### PR DESCRIPTION
**Pull request by darkskygit**

### CODE REVIEW

File: packages/backend/server/src/core/auth/controller.ts [hunk 1]
```diff
@@ -147,6 +152,33 @@ export class AuthController {
       if (!allowSignup) {
         throw new SignUpForbidden();
       }
+
+      const requireEmailDomainVerification = await this.config.runtime.fetch(
```
Comment (EYHN) [inline packages/backend/server/src/core/auth/controller.ts:162]: Will this cause registered users to be unable to log into their accounts?
We should only check this when registering

Comment (darkskygit) [reply packages/backend/server/src/core/auth/controller.ts:162]: this logic only execute that user not register yet

File: packages/backend/server/src/core/auth/controller.ts [hunk 2]
```diff
@@ -30,6 +32,9 @@ import { AuthService } from './service';
 import { CurrentUser, Session } from './session';
 import { TokenService, TokenType } from './token';
 
+// set DNS servers
+setServers(['1.1.1.1', '8.8.8.8']);
```
Comment (Brooooooklyn) [inline packages/backend/server/src/core/auth/controller.ts]: why we need this? Our Node.js instances are in k8s cluster, there may rpc requires `coredns` in the future, can we remove this?

Comment (darkskygit) [reply packages/backend/server/src/core/auth/controller.ts]: in the local dev env, if we enable network software like clash tun/surge/tailscale/zerotier, thire DNS hijacking logic may not correctly provide domain records other than DNS. let me add a conditional logic

Comment (darkskygit) [reply packages/backend/server/src/core/auth/controller.ts]: fixed

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**